activation/activation.go

```diff
@@ -96,6 +98,14 @@ type Builder struct {
 	stop          context.CancelFunc
 }
 
+type positioningAtxFinder struct {
+	finding sync.Mutex
+	found   *struct {
+		id         types.ATXID
+		forPublish types.EpochID
+	}
```

Comment:
I don't think caching the found ATX is needed? If calls to `Builder::getPositioningAtx` are protected by a mutex the second call should fetch the same ATX from the DB anyway and already know that that ATX is valid or not?

Reply:
The second call might find a different candidate ATX (and verify its chain) if new ATXs came in between the calls. It could happen if PoST prooving for one ID takes longer than the other. In my opinion, it should stick to the once-selected positioning ATX for all IDs (instead of always picking the best available at the current time) - it will be necessary for ATX merge anyway.

Reply:
All identities wait until 1 hour before the end of the cycle gap to select the positioning ATX anyway - so under normal circumstances (identities finish well within the time of the CG) they should all pick the positioning ATX at the same time.

> In my opinion, it should stick to the once-selected positioning ATX for all IDs (instead of always picking the best available at the current time) - it will be necessary for ATX merge anyway.

Identities that are excluded from a merged ATX (e.g. because they are late) will then also pick the same positioning ATX. This value is will be cached for 2 weeks - if one identity publishes in an early phase, the selected positioning ATX will always have a low height for all identities although they should pick they highest tick ATX they see at the time of registration.

Reply:
> under normal circumstances (identities finish well within the time of the CG) 

How do you decide these are normal circumstances? Every ID is allowed to finish proving PoST at any time within the cycle gap, including 1 minute before it ends. In some setups, that are utilizing DAG to schedule post-services in groups (i.e. ID by ID etc.), the last PoSTs might come on the boundary.

> Identities that are excluded from a merged ATX (e.g. because they are late) will then also pick the same positioning ATX. This value is will be cached for 2 weeks - if one identity publishes in an early phase, the selected positioning ATX will always have a low height for all identities although they should pick they highest tick ATX they see at the time of registration.

The client doesn't support multiple poet phases so this issue doesn't exist. If an identity misses a poet round it will sleep for an epoch and select its positioning ATX targeting the next publish epoch.

Supposing that we implement support for multiple phases, the caching is still valuable but the _key_ would need to be different (maybe a combination of `poet-phase-identifier+publish-epoch`).